### ISSUE TITLE
Migrate farmers from IYO to 3bot login

### ISSUE THREAD

**Issue by andhartl** [threefoldtech/tf_cockpit#12]
Document and plan migration of farmer logins to 3bot login. Make sure the registered nodes will not be affected.

**Comment by zaibon** [comment]
So the idea is that all farmers migrate from IYO to the 3bot login.

There are multiple things to migrate:
- The IYO user of the farmer itself
- The IYO organization that was used to identify the farm
- All the information about a farm (name, location, wallet addresses, email,...)

Before going any further I need to know how we want to identify a farm using the 3bot ID. Are we going to create a new identity or is the farm going to be a derivative identity from the farmer ID ?
@andhartl I need your input on this.

Once we know how we want to organize identity, the rest should be pretty straight forward.
Since both IYO and 3bot offer some kind of OAuth flow creating a small migration application that exchange information shouldn't be a problem.

## Required feature/code before we can do anything:
- [BCDB 3bot registry](https://github.com/threefoldtech/jumpscaleX_threebot/tree/master/ThreeBotPackages/threefold/phonebook): Gedis actor that allow us to save 3bot ID in BCDB
- [BCDB farm directory](https://github.com/threefoldtech/jumpscaleX_threebot/tree/master/ThreeBotPackages/threefold/directory): Gedis actor that allow us to save farm details

**Comment by andhartl** [comment]
@zaibon would reuse the existing farmer ids. I mean the 3bot account name should be the farmer id. reason: we do not have sub-orgs within a 3bot
makes sense?

**Comment by zaibon** [comment]
@andhartl nop. 
Farmer ID will for sure change. Reason is the actual farmer ID is a IYO JWT. This is not going to be something we can keep and I also don't want to keep it.
We are aiming to a more public key oriented identification system (which is good). So I want to leverage this everywhere in our ecosystem. So I really would love we have an unified way to identify something or someone.

As far as I understand how 3bot login works, is the public key of a user that is the used to identify it. So we need to apply the same concept to any ID we generate throughout the grid.
This will be the same for the node themself. They will have a key pair and be identified by their public key.

@jdelrue Can you comment on that.

**Comment by andhartl** [comment]
did  not mean the farmer id jwt token, mean the suborg name aka farmer id (organisation). see pic
<img width="1119" alt="image" src="https://user-images.githubusercontent.com/18591016/63790005-bd024d00-c8f8-11e9-94c0-9c14e1b6d95f.png">

**Comment by zaibon** [comment]
ha ok got it know. Well yes I think these 2 maps pretty good.
We probably get rid of the distinction between `Name` and `Farmer ID`, just have one an make it unique. Now the question remains, how are we going to link a farm to a user (or group of users) ? As far as I know there is not such relation modeled in the 3bot app yet

**Comment by andhartl** [comment]
would say let's use the 'Farmer ID' as the '3Bot name'. this would make the migration for existing farmers easier. Skip the 'Name' for sure. @jdelrue can '3bot name' be with ' _'?

**Comment by jdelrue** [comment]
We don't have multiple accounts yet, we have no such thing as organizations. Is a farm managed by multiple users? Which functionality do we need and have at the moment? Can we get a demo?

**Comment by zaibon** [comment]
@jdelrue 
currently in the v1 of the grid here is how things were working:

A user is registered on IYO. If he wanted to become a farmer, he needed to create an organization on IYO and then register a new farm on the directory website: https://capacity.threefoldtoken.com/register_farm

Registering a farm generates a farmerID (IYO JWT) that is required to be configured on the node, so they know to which farmer they belong. This information is required to be able to pay farmers.

Now since we move away from IYO. We need to redesign this. 
As far as I know there is nothing set in stone so far so we have some room here.

Regarding the demo, not sure what you want to see?

**Comment by zaibon** [comment]
@jdelrue could we imagine a system where a farm ID is somehow a derivative from the user ID that owns it ? Or that wouldn't bring anything ? I'm trying to come up with a scheme where we wouldn't have to keep too much data in a database to actually link the two identity but instead just use cryptography for it. So it is built-in the identity itself.

**Comment by jdelrue** [comment]
Yes we can make derivative keys, we use it for the wallet. But its not shared among users.

**Comment by serboctor** [comment]
I was working on a migration actor as part of this task. https://github.com/threefoldtech/jumpscaleX_threebot/issues/23 
Where has the discussion reached in regard to threebot_id for migrated farms?

**Comment by jdelrue** [comment]
> That looks about right.
> Here is the URL of the API of the old directory where you can get the defailt of a legacy farm:
> `https://capacity.threefoldtoken.com/api/farmers/{iyo_organization}`
